### DISCUSSION TITLE
Is there a way to evaluate a chunk conditional whether light or dark mode is activated?

### DISCUSSION THREAD

**Discussion by EllaKaye** [quarto-dev/quarto-cli#7713] [Q&A] [answered]
### Description

On one page of a Quarto website, I'd like to override my site-wide `custom-light.scss` and `custom-dark.scss` files to change the colour of links on hover. For one mode, this is easy to do with a css chunk on that page for a specified colour, e.g.

````
```{css}
a:hover {
  color: #006AD4;
}
```
````

However, I want the colour to be different depending on whether the website is in light or dark mode. Is there a way of doing that? Is it possible, for example, to evaluate such chunks depending on which mode is active? If this is not possible within a chunk, is there a way of doing it with a raw html block?

Even better, is there a way to do this that picks up variables in my `custom-light.scss` and `custom-dark.scss` files. It would be fantastic to be able to do something like:

````
```{scss}
a:hover {
  color: $second-color;
}
```
````

where `$second-color` is set to different colours in my light and dark theme scss files.

I have also tried defining additional `extra-light.scss` and `extra-dark.scss` files, where I define a `$second-color` variable in each, then linking to those in the page yaml, but that doesn't work:

e.g. in `extra-light.scss` (and similar for `extra-dark.scss` with a different color):

```
$second-color: #006AD4; 

a:hover {
    color: $second-color;
}
```

and in the page yaml:

```
theme:
  light: extra-light.scss
  dark: extra-dark.scss
```

**Comment by jimjam-slam** [comment]
The `<body>` has a class of `quarto-light` or `quarto-dark`, so it might be easier to use those in conjunction with your selector!

```scss
body.quarto-dark a:hover {
  color: $second-color;
}
```

**Reply by mcanouil** [reply]
Indeed, this would be the more reliable way to have conditional (S)CSS based on themes.

**Comment by EllaKaye** [comment, marked as answer]
@jimjam-slam, that's a much neater solution for what I wanted to achieve. Thank you 😄. I wasn't sure about making the page-specific chunk pick up the scss variable, but hard-coding the colours is good enough. So now, this css chunk is working well:

````
```{css}
body.quarto-light a:hover {
  color: #006AD4; 
}

body.quarto-dark a:hover {
  color: #D46A00; 
}
```
````

**Reply by mcanouil** [reply]
@EllaKaye SCSS is a compiled version of CSS, so you can't add SCSS outside of the compilation path which is in Quarto via the `theme` key.
The compilation then creates a static CSS that is used on the website.
So in your case, for a specific change, you can't use SCSS (shared everywhere), but have to use CSS.
Note that it is a bit clearer to include the CSS via `include-in-header` rather than as content via a code cell.

**Reply by EllaKaye** [reply]
Thanks @mcanouil. I can't seem to make `include-in-header` work for css. I've tried created `second-color.html` with 

```
<style>
body.quarto-light a:hover {
  color: #006AD4; 
}

body.quarto-dark a:hover {
  color: #D46A00; 
}
</style>
```

then in the page yaml:

```
include-in-header:
  file: second-color.html
```

but when I render, I just see the contents of `second-color.html` as text at the top of the page. What's the fix?

**Reply by mcanouil** [reply]
Here is a full example:

````qmd
---
title: "Quarto Playground"
format:
  html:
    theme:
      light: flatly
      dark: darkly
include-in-header:
  text: |
    <style type="text/css">
    body.quarto-light a:hover {
      color: #006AD4; 
    }

    body.quarto-dark a:hover {
      color: #D46A00; 
    }
    </style>
---

This is a playground for [Quarto](https://quarto.org).
````

**Reply by EllaKaye** [reply]
Thanks @mcanouil, so helpful (as always!) I didn't know about type="text/css". Using include-in-header with this has really tidied up my code. 

In case you'd like to see what this is for, it's now live at <https://adventofcode.ellakaye.co.uk>. On each year's listing page, the post titles (on hover) match the colour of the page's title and navbar link. It's only a small detail, but I think it's pretty neat!

**Reply by mcanouil** [reply]
The type attribute is not mandatory, this is also true for script tag, but it makes everything explicit.
Also for style, there is only one possible value 😅

https://www.w3schools.com/tags/att_style_type.asp
